storm-core/src/jvm/org/apache/storm/trident/Stream.java

```diff
@@ -124,6 +124,31 @@ public Stream parallelismHint(int hint) {
     }
 
     /**
+     * Sets the CPU Load resource for the current node
+     */
+    public Stream setCPULoad(Number load) {
```

Comment:
A `Stream` should ideally have only streaming operations but it seems we are bringing the topology configuration into the stream api which does not look very clean. Cant this be part of the `TridentTopology` or passed during `submitTopology` ? If this should really be part of stream, can we have a single method like `nodeConfig(NodeConfig config)` and have all these methods as part of the `NodeConfig` so that we have only one extra method which is not really a stream operation?